Packages/src/TypeScriptServer~/src/mcp-client-compatibility.ts

```
@@ -2,34 +1,47 @@
+ * - Handles Unity client name registration and reconnection scenarios
+ * - Provides logging for debugging compatibility issues
  */
 export class McpClientCompatibility {
-  private unityClient: UnityClient;
   private clientName: string = DEFAULT_CLIENT_NAME;
 
-  constructor(unityClient: UnityClient) {
-    this.unityClient = unityClient;
+  constructor() {
+    // UnityClient dependency removed - setClientName now handled by ClientInitializationUseCase
   }
```

Biome (1.9.4)

[error] This constructor is unnecessary.

Unsafe fix: Remove the unnecessary constructor.


(lint/complexity/noUselessConstructor)